File: app/src/main/java/com/likhit/variants/data/models/BaseResponse.java
package com.likhit.variants.data.models;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class BaseResponse implements Serializable {

    @SerializedName("variants")
    private Variants variants;

    public BaseResponse(Variants variants) {
        this.variants = variants;
    }

    public BaseResponse(BaseResponse response) {
        this.variants = new Variants(response.getVariants());
    }

    public Variants getVariants() {
        return variants;
    }
}

File: app/src/main/java/com/likhit/variants/data/models/BaseResponse.kt
package com.likhit.variants.data.models

import com.google.gson.annotations.SerializedName

import java.io.Serializable

class BaseResponse : Serializable {

    @SerializedName("variants")
    var variants: Variants? = null
        private set

    constructor(variants: Variants) {
        this.variants = variants
    }

    constructor(response: BaseResponse) {
        this.variants = Variants(response.variants!!)
    }
}
